### PR TITLE
"Freed the blocks from the evil CompatLayer!"

### PR DESCRIPTION
**Pull request by azuraerae**
- Fixed various doubles in translation file.

- Gave all the blocks proper registryNames (i.e copper_ore) and changed en_US.lang and all textures/blockstates accordingly.

- Made two Block constructors BlockBase (and ItemBase) and OreBase with according functions. (Drop chance, dropped item for ease of access.)

- Switched the block registration method from the Compatibility1_12_2 layer to the default e.getRegistry() registration method. (Switching back to Forge's default.) -> Made a proper init package, with Blocks (and Items) classes to make proper use of the @SubscribeEvent flags. -> Replaced all Block calls from the Ores class to the new Blocks class.

-> Planning on doing the same for Items (and eventually getting rid of the Ores class alltogether).

- Raeilgun (Azurae)

### CODE REVIEW

File: src/main/java/com/paneedah/mwc/RecipeManager.java [hunk 1]
```diff
@@ -12,9 +13,9 @@ public class RecipeManager {
 
     public static void init(ModContext modContext) {
 
-        compatibility.addSmelting(Ores.CopperOre, new ItemStack(Ores.CopperIngot), 5f);
-        compatibility.addSmelting(Ores.LeadOre, new ItemStack(Ores.LeadIngot), 5f);
-        compatibility.addSmelting(Ores.TinOre, new ItemStack(Ores.TinIngot), 5f);
+        compatibility.addSmelting(Blocks.copper_ore, new ItemStack(Ores.CopperIngot), 5f);
+        compatibility.addSmelting(Blocks.lead_ore, new ItemStack(Ores.LeadIngot), 5f);
+        compatibility.addSmelting(Blocks.tin_ore, new ItemStack(Ores.TinIngot), 5f);
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/RecipeManager.java:18]: Need to revert that this isn't the Minecraft id this is just the variable should follow Java convention

File: src/main/java/com/paneedah/mwc/bases/BlockBase.java [hunk 2]
```diff
@@ -1,31 +1,28 @@
...
+        setRegistryName(registryName);
+        setTranslationKey(registryName);
 
-        setTranslationKey(ModReference.id + "_" + "GraphiteOre");
-        //setBlockTextureName(ModReference.id + ":" + "graphiteore");
         setHardness(6F);
         setResistance(15F);
-        setSoundType(SoundType.STONE); 
-        setHarvestLevel("pickaxe", 2);
+        setHarvestLevel("pickaxe",2);
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/BlockBase.java:19]: Missing the space

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.Item;
+
+import java.util.Random;
+
+public class OreBase extends BlockOre {
+    Item itemBlock, drop;
+    int harvestLevel = 2, minDrop = 1, maxDrop = 1;
+    boolean smelt = true;
+    public OreBase(String registryName){
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:15]: Missing space before bracket

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import com.paneedah.mwc.ModernWarfareMod;
+import com.paneedah.mwc.utils.ModReference;
+import net.minecraft.block.BlockOre;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.Item;
+
+import java.util.Random;
+
+public class OreBase extends BlockOre {
+    Item itemBlock, drop;
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:12]: Missing space between class init and variables

File: src/main/java/com/paneedah/mwc/bases/OreBase.java (hunk 3 above)
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:15]: Missing space between method and variable

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        itemBlock = Item.getItemFromBlock(this);
+        //If the block is smelt-able (i.e like gold or iron) then drop the block itself, otherwise drop the designated drop.
+        if(this.smelt) {
+            if(this.drop != null) {
+                ModReference.log.warn("Block " + this.getRegistryName() + " is smeltable but does not drop itself!");
+            }
+            return itemBlock;
+        } else {
+            return this.drop;
+        }
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:36]: Single line if block should not have brackets

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Override
+    public int quantityDropped(Random random) {
+        int i = 0;
+
+        if(this.maxDrop - this.minDrop > 0){
+            i = random.nextInt(this.maxDrop - this.minDrop);
+        }
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:45]: Single line if block should not have brackets

**Comment by Paneedah** [comment]
All good on my end! Waiting for @JustDesoroxxx to review before merge.

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    public static OreBase[] blocks;
+
+    public static void init(){
+        copper_ore = new OreBase("copper_ore");
+
+        tin_ore = new OreBase("tin_ore");
+
+        lead_ore = new OreBase("lead_ore");
+
+        sulfur_ore = new OreBase("sulfur_ore");
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:35]: this should probably be close together

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        sulfur_ore.isSmeltable(false);
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:36]: Missing space between two different actions

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        sulfur_ore.setItemDropped(Ores.SulfurDust);
+        sulfur_ore.setDropAmount(2, 5);
+
+        graphite_ore = new OreBase("graphite_ore");
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:40]: Should be with the others

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                lead_ore,
+                graphite_ore
+        };
+    }
+
+    @SubscribeEvent
+    public static void registerBlocks(RegistryEvent.Register<Block> e){
+        init();
+        e.getRegistry().registerAll(blocks);
+    }
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:58]: `e` should be `blockRegistryEvent`

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    public static void registerItemBlock(RegistryEvent.Register<Item> e){
+        Item[] items = new Item[blocks.length];
+
+        for (int i = 0; i < blocks.length; i++) {
+            items[i] = new ItemBlock(blocks[i]);
+            items[i].setRegistryName(blocks[i].getRegistryName());
+        }
+
+        e.getRegistry().registerAll(items);
+    }
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:70]: `e` should be `itemRegistryEvent`

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+    @SubscribeEvent
+    public static void registerRenders(ModelRegistryEvent e){
+        for (Block block : blocks){
+            if(block instanceof BlockBase || block instanceof OreBase){
+                ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(block), 0, new ModelResourceLocation(block.getRegistryName().toString(), "inventory"));
+            }
+        }
+    }
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:79]: `e` should be `modelRegistryEvent`

File: src/main/java/com/paneedah/mwc/init/Items.java [hunk 13]
```diff
@@ -0,0 +1,13 @@
+package com.paneedah.mwc.init;
+
+import com.paneedah.mwc.bases.ItemBase;
+import com.paneedah.mwc.utils.ModReference;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod.EventBusSubscriber(modid = ModReference.id)
+public class Items {
+    public static ItemBase sulfur_dust = new ItemBase();
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/init/Items.java:9]: Missing space between class init and actuall stuff

File: src/main/java/com/paneedah/mwc/bases/ItemBase.java [hunk 14]
```diff
@@ -0,0 +1,9 @@
+package com.paneedah.mwc.bases;
+
+import net.minecraft.item.Item;
+
+public class ItemBase extends Item {
+    public ItemBase(){
```
Comment (JustDesoroxxx) [inline src/main/java/com/paneedah/mwc/bases/ItemBase.java:6]: Missing space between class init and actuall stuff

**Review by JustDesoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/mwc/RecipeManager.java (hunk 1 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/RecipeManager.java:18]: ✅

File: src/main/java/com/paneedah/mwc/bases/BlockBase.java (hunk 2 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/bases/BlockBase.java:19]: ✅

File: src/main/java/com/paneedah/mwc/bases/OreBase.java (hunk 3 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/bases/OreBase.java:15]: ✅

File: src/main/java/com/paneedah/mwc/bases/OreBase.java (hunk 4 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/bases/OreBase.java:12]: ✅

File: src/main/java/com/paneedah/mwc/bases/OreBase.java (hunk 3 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/bases/OreBase.java:15]: ✅

File: src/main/java/com/paneedah/mwc/init/Blocks.java (hunk 10 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/init/Blocks.java:58]: ✅

File: src/main/java/com/paneedah/mwc/init/Blocks.java (hunk 11 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/init/Blocks.java:70]: ✅

File: src/main/java/com/paneedah/mwc/init/Items.java (hunk 13 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/init/Items.java:9]: ✅

File: src/main/java/com/paneedah/mwc/bases/ItemBase.java (hunk 14 above)
Comment (azuraerae) [reply src/main/java/com/paneedah/mwc/bases/ItemBase.java:6]: ✅

File: src/main/java/com/paneedah/mwc/bases/BlockBase.java [hunk 15]
```diff
@@ -1,31 +1,28 @@
...
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.item.Item;
 
 import java.util.Random;
 
-public class BlockGraphiteOre extends Block {
-    
-    public BlockGraphiteOre() {
+public class BlockBase extends Block {
+    public BlockBase(String registryName) {
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/bases/BlockBase.java:12]: Missing space between class init and actuall stuff

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+        itemBlock = Item.getItemFromBlock(this);
+        //If the block is smelt-able (i.e like gold or iron) then drop the block itself, otherwise drop the designated drop.
+        if(this.smelt) {
+            if(this.drop != null) {
+                ModReference.log.warn("Block " + this.getRegistryName() + " is smeltable but does not drop itself!");
+            }
+            return itemBlock;
+        } else {
+            return this.drop;
+        }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:37]: Single line if block should not have brackets

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    public int quantityDropped(Random random) {
+        int i = 0;
+
+        if(this.maxDrop - this.minDrop > 0){
+            i = random.nextInt(this.maxDrop - this.minDrop);
+        }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:46]: Single line if block should not have brackets

File: src/main/java/com/paneedah/mwc/init/Items.java [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.paneedah.mwc.bases.ItemBase;
+import com.paneedah.mwc.utils.ModReference;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod.EventBusSubscriber(modid = ModReference.id)
+public class Items {
+
+    public static ItemBase sulfur_dust = new ItemBase();
+    public static void init(){
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/init/Items.java:11]: Missing space between variable and method

File: src/main/java/com/paneedah/mwc/init/Blocks.java [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+
+        itemRegistryEvent.getRegistry().registerAll(items);
+    }
+
+    @SubscribeEvent
+    public static void registerRenders(ModelRegistryEvent modelRegistryEvent){
+        for (Block block : blocks){
+            if(block instanceof BlockBase || block instanceof OreBase){
+                ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(block), 0, new ModelResourceLocation(block.getRegistryName().toString(), "inventory"));
+            }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/init/Blocks.java:77]: Single line if block should not have brackets

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/mwc/bases/OreBase.java [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+import net.minecraft.item.Item;
+
+import java.util.Random;
+
+public class OreBase extends BlockOre {
+
+    Item itemBlock, drop;
+    int harvestLevel = 2, minDrop = 1, maxDrop = 1;
+    boolean smelt = true;
+    public OreBase(String registryName) {
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/bases/OreBase.java:16]: Missing space between variables and method

File: src/main/java/com/paneedah/mwc/init/Items.java [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.paneedah.mwc.bases.ItemBase;
+import com.paneedah.mwc.utils.ModReference;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod.EventBusSubscriber(modid = ModReference.id)
+public class Items {
+
+    public static ItemBase sulfur_dust = new ItemBase();
+    public static void init() {
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/init/Items.java:11]: Missing space between variables and method

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/mwc/bases/BlockBase.java (hunk 15 above)
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/bases/BlockBase.java:12]: Missing space between class initialization and method

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

**Review by Desoroxxx — APPROVED** [review]

**Pull request merged**